src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderConfig.tsx

```diff
@@ -276,41 +236,177 @@ export const LimitOrderConfig = ({
     translate,
   ])
 
+  const displayAsset = useMemo(() => {
+    return priceDirection === PriceDirection.BuyAssetDenomination ? sellAsset : buyAsset
+  }, [buyAsset, priceDirection, sellAsset])
+
+  const handleTokenTextClick = useCallback(() => {
+    handleTogglePriceDirection()
+  }, [handleTogglePriceDirection])
+
+  const sellAmountCryptoPrecision = useAppSelector(selectInputSellAmountCryptoPrecision)
+
+  const handleToggleFiatSellAmount = useCallback(() => {
+    onChangeIsInputtingFiatSellAmount(!isInputtingFiatSellAmount)
+  }, [isInputtingFiatSellAmount, onChangeIsInputtingFiatSellAmount])
+
+  const handleInputValueChange = useCallback(() => {
+    handleInputChange(priceAmountRef.current ?? '0')
+  }, [handleInputChange])
+
+  const receiveAmount = useMemo(() => {
+    if (
+      bnOrZero(sellAmountCryptoPrecision).isZero() ||
+      bnOrZero(limitPrice.buyAssetDenomination).isZero()
+    ) {
+      return '0'
+    }
+
+    if (priceDirection === PriceDirection.BuyAssetDenomination) {
+      return bnOrZero(sellAmountCryptoPrecision).times(limitPrice.buyAssetDenomination).toString()
+    }
+
+    return bnOrZero(sellAmountCryptoPrecision).div(limitPrice.sellAssetDenomination).toString()
+  }, [limitPrice, priceDirection, sellAmountCryptoPrecision])
+
+  const limitOrderExplanation = useMemo(() => {
+    if (
+      isLoading ||
+      bnOrZero(sellAmountCryptoPrecision).isZero() ||
+      bnOrZero(limitPriceForSelectedPriceDirection).isZero()
+    )
+      return null
+
+    const formattedPrice = bnOrZero(limitPriceForSelectedPriceDirection).toFixed(6)
+    const formattedSellAmount = bnOrZero(sellAmountCryptoPrecision).toFixed(6)
+    const formattedReceiveAmount = bnOrZero(receiveAmount).toFixed(6)
```

Comment:
preferably-blocking: all of these should use programmatic precision, and trim zeros